### PR TITLE
Update submodule tags to pass runoff from CISM to ROF as per issue #2590

### LINKED ISSUE #2590: Update CMEPS/MOSART/CISM/RTM tags
Update MOSART, CMEPS and CISM so CISM runoff goes to ROF rather than CTSM
Update RTM with fix needed for Paleo LGM work

### PR DESCRIPTION
**Pull request by slevis-lmwg**
### Description of changes
Details in #2590 

### Specific notes

Contributors other than yourself, if any:
@ekluzek @mvertens @jedwards4b @billsacks @Katetc 

CTSM Issues Fixed (include github issue #):
Fixes #2590 

Are answers expected to change (and if so in what way)?
Yes, due to change in default behavior

Any User Interface Changes (namelist or namelist defaults changes)?
Default behavior has changed as explained in #2590 

Does this create a need to change or add documentation? Did you do so?
Probably, but I did not.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -7,8 +7,8 @@ fxDONOTUSEurl = https://github.com/NCAR/fates-release
 
 [submodule "cism"]
 path = components/cism
-url = https://github.com/mvertens/CISM-wrapper
-fxtag = 34339777afa7b810fd4822fce1039826668b33e4
+url = https://github.com/ESCOMP/CISM-wrapper
+fxtag = cismwrap_2_2_001
```
Comment (slevis-lmwg) [inline .gitmodules:11]: I'm puzzled about why these changes appear in this commit but do not appear when I "show all changes"

Comment (slevis-lmwg) [reply .gitmodules:11]: ...because this is not the new tag :-)

**Comment by slevis-lmwg** [comment]
From the [TODO list in the issue](https://github.com/ESCOMP/CTSM/issues/2590#issuecomment-2179451664)
Most of aux_clm and ctsm_sci is failing for now.

**Comment by slevis-lmwg** [comment]
Bisecting between my passing and failing aux_clm test-suites:
- no code diffs between rtm1_0_80 and 420c5979d8abf05184992fdfba5ac41c41c91c2e
- no code diffs between mosart1.1.02 and ca0acc0336365456bec653c4c233abb57d9bc62d
**- large code diffs between cmeps0.14.67 and 4dca7c602e54189b74b1d0195c433d1b83475553
Here https://github.com/ESCOMP/CMEPS/commit/f960f3ed3f4edc6fd59bbfe171160e179557062a I see that I may have had to have used cmeps0.14.77**
@jedwards4b I should have checked with you which tag to use, for this one and the next one. I have resubmitted aux_clm on izumi already, so I hope to know by tomorrow, but pls let me know if you think that I'm still off.
- ccs_config now I see here https://github.com/ESMCI/ccs_config_cesm/commit/4e337b5e7af44e0ac7226a83e4349154 that I may have had to have used ccs_config_cesm1.0.0
- cism should be fine because I asked @Katetc specifically which tag to use

**Comment by slevis-lmwg** [comment]
aux_clm on izumi now OK
`./cs.status.fails | grep -v 'otherwise bit-for-b' | grep -v NLCOMP | grep -v 'EXPECTED FAILURE' | grep -v PASS`

on derecho: waiting for the machine to become available.

**Comment by slevis-lmwg** [comment]
@swensosc verified by email that these diffs seem reasonable. Sean wrote:
I think you would see changes in methane due to the fact that the riley et al. method for calculating inundation uses a regression against TWS to determine the inundated fraction, which is an input to the methane model.  So if VOLR is changed, TWS would be changed, and thus methane would see a perturbed inundated area.

**Comment by slevis-lmwg** [comment]
testing on derecho
PASS ./build-namelist_test.pl
PASS make black and make lint
PASS ./run_ctsm_py_tests -u and ./run_ctsm_py_tests -s

on derecho and izumi
IN PROG `./run_sys_tests -s aux_clm -c ctsm5.2.014 -g ctsm5.2.015`

**Comment by slevis-lmwg** [comment]
@jedwards4b I updated to ctsm5.2.014 and submitted tests as posted above. If all goes smoothly, I should merge this PR next.

First I wanted your input on something that seems concerning to me about ccs_config:
`./bin/git-fleximod update` (with or without --force) returns...
```
s  ccs_config ccs_config_cesm0.0.113 69a958581ecd2d32ee9cb1c38bcd3847b8b920bf is out of sync with .gitmodules ccs_config_cesm1.0.0

ccs_config updated to ccs_config_cesm1.0.0
```
and then status returns...
`s  ccs_config ccs_config_cesm0.0.113 69a958581ecd2d32ee9cb1c38bcd3847b8b920bf is out of sync with .gitmodules ccs_config_cesm1.0.0`
and git describe returns...
`ccs_config_cesm0.0.113`
while .gitmodules says...
`fxtag = ccs_config_cesm1.0.0`
What do you think?

**Comment by slevis-lmwg** [comment]
@jedwards4b 
I tried `git diff ccs_config_cesm1.0.0 ccs_config_cesm0.0.113`
and got no difference, so I will assume that ccs_config is acceptable for the purposes of this PR's testing.

**Review by jedwards4b — APPROVED** [review]

**Comment by slevis-lmwg** [comment]
Thank you @jedwards4b 

aux_clm results
`./cs.status.fails | grep -v 'ise bit-for-b' | grep -v NLCOMP|grep -v PASS | grep -v '14: DIF'`
OK izumi
OK derecho
Last TODO before merge:
I want to double check the nature of the DIFFs from baseline.

**Comment by slevis-lmwg** [comment]
These differences seem out of the ordinary:
```
SMS_D.1x1_brazil.I2000Clm60FatesSpCruRsGs.derecho_gnu.clm-FatesColdDryDepSatPhen.GC.0719-175606de_gnu/SMS_D.1x1_brazil.I2000Clm60FatesSpCruRsGs.derecho_gnu.clm-FatesColdDryDepSatPhen.GC.0719-175606de_gnu.clm2.h0.2000-01-01-00000.nc.cprnc.out: RMS FATES_TRANSITION_MATRIX_LULU       Infinity            NORMALIZED    Infinity
SMS_D.1x1_brazil.I2000Clm60FatesSpCruRsGs.derecho_gnu.clm-FatesColdMeganSatPhen.GC.0719-175606de_gnu/SMS_D.1x1_brazil.I2000Clm60FatesSpCruRsGs.derecho_gnu.clm-FatesColdMeganSatPhen.GC.0719-175606de_gnu.clm2.h0.2000-01-01-00000.nc.cprnc.out: RMS FATES_TRANSITION_MATRIX_LULU       Infinity            NORMALIZED    Infinity
```
@glemieux are these new tests that came with your recent PR? Does this concern you?

**Comment by glemieux** [comment]
> These differences seem out of the ordinary:
> 
> ```
> SMS_D.1x1_brazil.I2000Clm60FatesSpCruRsGs.derecho_gnu.clm-FatesColdDryDepSatPhen.GC.0719-175606de_gnu/SMS_D.1x1_brazil.I2000Clm60FatesSpCruRsGs.derecho_gnu.clm-FatesColdDryDepSatPhen.GC.0719-175606de_gnu.clm2.h0.2000-01-01-00000.nc.cprnc.out: RMS FATES_TRANSITION_MATRIX_LULU       Infinity            NORMALIZED    Infinity
> SMS_D.1x1_brazil.I2000Clm60FatesSpCruRsGs.derecho_gnu.clm-FatesColdMeganSatPhen.GC.0719-175606de_gnu/SMS_D.1x1_brazil.I2000Clm60FatesSpCruRsGs.derecho_gnu.clm-FatesColdMeganSatPhen.GC.0719-175606de_gnu.clm2.h0.2000-01-01-00000.nc.cprnc.out: RMS FATES_TRANSITION_MATRIX_LULU       Infinity            NORMALIZED    Infinity
> ```
> 
> @glemieux are these new tests that came with your recent PR? Does this concern you?

@slevis-lmwg can you point me to the test folder?  I'd like to check out the `cprnc` output.

**Comment by slevis-lmwg** [comment]
@glemieux please look here
/glade/work/slevis/git_externals/cism2mosart/tests_0719-175606de

**Comment by jedwards4b** [comment]
There is a permissions problem for your baseline directory.  
ls: cannot open directory '/glade/campaign/cgd/tss/ctsm_baselines/ctsm5.2.014/': Permission denied

**Comment by slevis-lmwg** [comment]
@jedwards4b I updated permissions for that one and will go through and update for my other ones, too. LMK if you still have trouble accessing.

**Comment by jedwards4b** [comment]
Thank you for fixing that - the values for the field FATES_TRANSITION_MATRIX_LULU in both the baseline and the case look like garbage to me.  Can someone confirm these values are expected?

**Comment by slevis-lmwg** [comment]
@glemieux is taking a look.

**Comment by glemieux** [comment]
Yep, those cases are definitely garbage, but they've been that way for these two test since that variable was introduced in `ctsm5.2.013`.  I missed this during testing.  Those values should be zero'd out.  I'm not sure why the `FatesColdSatPhen` correctly zeros `FATES_TRANSITION_MATRIX_LULU`, but `FatesColdDryDepSatPhen` and `FatesColdMeganSatPhen` do not.  

I'd say this is ok as is to go forward, but an issue should be created to fix this as it will likely flag again given the randomness of the garbage.

**Comment by slevis-lmwg** [comment]
I'm fine with that. I can open the issue and you can add to it as you see fit @glemieux 
Thanks for looking into this.
